### PR TITLE
Add support to lookup of extensions

### PR DESCRIPTION
**Pull request by laeubi**
This add first support of lookup extension components, currently only
project scoped extensions are supported but could be enhanced later to
support core extensions as well.

### CODE REVIEW

**Comment by laeubi** [comment]
@kjsmita6 this should now be the "dynamic" variant of your change, do you think you can test this with your custom plugin as well? With tycho extension it looks like this:

grafik

**Comment by laeubi** [comment]
@mickaelistria any concerns?

**Comment by kjsmita6** [comment]
@laeubi Looks good with my testing (checked out your fork). In my pom I have defined:
```xml
  <build>
        <plugins>
            <plugin>
                <groupId>com.tibco.ep</groupId>
                <artifactId>ep-maven-plugin</artifactId>
                <extensions>true</extensions>
            </plugin>
        </plugins>
    </build>
```

And shows up in the pom editor as:
Screenshot (57)

File: org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/ExtensionsTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.eclipse.m2e.core;
```
Comment (mickaelistria) [inline org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/ExtensionsTest.java:1]: Please include legal header.

**Review by mickaelistria — CHANGES_REQUESTED** [review]

**Comment by mickaelistria** [comment]
@laeubi I'd like to merge it ASAP (only concern is the missing copyright header on new file) as this improves an important part of m2e and really emphasizes the progress that are possible with recent changes.

File: org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/ExtensionsTest.java (hunk 1 above)
Comment (laeubi) [reply org.eclipse.m2e.core.tests/src/org/eclipse/m2e/core/ExtensionsTest.java:10]: Done.

**Comment by laeubi** [comment]
I have added the header now.

**Comment by mickaelistria** [comment]
Thanks, that's a great patch!

**Comment by laeubi** [comment]
> Thanks, that's a great patch!

Its just a first small step so don't be to enthusiastic :-)

**Pull request merged**